stdlib/public/Differentiation/ArrayDifferentiation.swift

```diff
@@ -218,10 +219,13 @@ extension Array where Element: Differentiable {
   ) {
     func pullback(_ v: TangentVector) -> (TangentVector, TangentVector) {
       precondition(
-        v.base.count == lhs.count + rhs.count, """
-          Tangent vector with invalid count; expected to equal the sum of \
-          operand counts \(lhs.count) and \(rhs.count)
+        v.base.isEmpty || v.base.count == lhs.count + rhs.count, """
+          Tangent vector with invalid count \(v.base.count); expected to \
+          equal the sum of operand counts \(lhs.count) and \(rhs.count)
           """)
+      if v.base.isEmpty {
```

Comment:
If you move the `if` block before the `precondition`, the precondition won't need to be changed.

Reply:
Ok.